iroh-sync/src/store/memory.rs

```diff
@@ -426,10 +426,7 @@ impl crate::ranger::Store<RecordIdentifier, SignedEntry> for ReplicaStoreInstanc
         Ok(self.with_records(|records| {
             records.and_then(|r| {
                 let v = r.get(&(key.author(), key.key().to_vec()))?;
-                if v.timestamp() == key.timestamp() {
-                    return Some(v.clone());
-                }
-                None
+                Some(v.clone())
```

Comment:
I really dislike this current interface but haven't found a good alternative. I think it this be a source of errors though, if we return values without checking that the timestamp matches

Reply:
You mean this flow:
* Insert entry `e1`. Clone the resulting `RecordIdentifier`, `r1`
* Insert entry `e2` with identical author and key.
* Get entry `r1`. You get `e2` while you expected to get `e1`?

I'm not sure. Now that there's now history ever, I would say this is expected behavior. Or did you mean something else?

Reply:
Actually, should we move the timestamp back into the `Record`?
It was moved from `Record` to `RecordIdentifier` in #1430 - but maybe it makes more sense semantically to have them in the `Record`. We have both record and id available at all places that needs them anyway.

Reply:
conceptually the timestamp really should be in the `Record`, so yes if we can do it such that we move it back that would be the best case imo